src/signals-unix.c

```diff
@@ -1033,6 +1033,10 @@ static void *signal_listener(void *arg)
         }
 #endif
         if (doexit) {
+            // Let's forbid threads from running GC while we're trying to exit,
+            // also let's make sure we're not in the middle of GC.
+            jl_atomic_fetch_add(&jl_gc_disable_counter, 1);
+            jl_safepoint_wait_gc(NULL);
```

Comment:
I don't really like making it wait for GC to continue from here, since it prevents the subsequent logic here from enabling the more aggressive exits on repeat signals. Could you move this to before the `jl_exit_thread0` call instead? The jl_exit_thread0_cb can then also re-enable GC.

For really fixing this though, we'd need to finish the PR for user-handling SIGINT and allow extending that to any of the 3 common control signals (SIGHUP 1, SIGINT 2, SIGTERM 15).

Reply:
Done.